jdaviz/configs/specviz/plugins/unit_conversion/tests/test_unit_conversion.py

```diff
@@ -1,92 +1,247 @@
+    converted_spectrum = conv_func(specviz_app.app, spectrum=spectrum1d,
                                    new_spectral_axis=new_spectral_axis)
 
-    result_spectral_axis = [0, 1.11111111, 2.22222222, 3.33333333,
-                            4.44444444, 5.55555556, 6.66666667, 7.77777778,
-                            8.88888889, 10]
+    assert_quantity_allclose(converted_spectrum.spectral_axis,
+                             RESULT_SPECTRAL_AXIS, atol=1e-5*u.um)
+    assert converted_spectrum.flux.unit == spectrum1d.flux.unit
+    assert converted_spectrum.spectral_axis.unit == new_spectral_axis
+    assert converted_spectrum.uncertainty is None
+
+
+def test_no_spec_no_flux_uncert(specviz_app, spectrum1d):
+    label = "Test 1D Spectrum"
+    specviz_app.load_spectrum(spectrum1d, data_label=label)
+
+    conv_func = uc.UnitConversion.process_unit_conversion
+    converted_spectrum = conv_func(specviz_app.app, spectrum=spectrum1d)
+
+    assert converted_spectrum.flux.unit == spectrum1d.flux.unit
+
+
+def test_no_spec_no_flux_uncert_unit_exp_none(specviz_app, spectrum1d):
+    np.random.seed(42)
+    label = "Test 1D Spectrum"
+    specviz_app.load_spectrum(spectrum1d, data_label=label)
+
+    spectrum1d.uncertainty = UnknownUncertainty(np.abs(
+        np.random.randn(len(spectrum1d.spectral_axis))))
+
+    conv_func = uc.UnitConversion.process_unit_conversion
+    converted_spectrum = conv_func(specviz_app.app, spectrum=spectrum1d)
+
+    assert converted_spectrum.flux.unit == spectrum1d.flux.unit
+    assert converted_spectrum.spectral_axis.unit == spectrum1d.spectral_axis.unit
+    assert converted_spectrum.uncertainty is None
+
+
+def test_no_spec_flux_no_uncert(specviz_app, spectrum1d):
+    label = "Test 1D Spectrum"
+    specviz_app.load_spectrum(spectrum1d, data_label=label)
+
+    spectrum1d.uncertainty = None
+    new_flux = "erg / (s cm2 um)"
+
+    conv_func = uc.UnitConversion.process_unit_conversion
+    converted_spectrum = conv_func(specviz_app.app, spectrum=spectrum1d,
+                                   new_flux=new_flux)
+
+    assert_quantity_allclose(converted_spectrum.flux,
+                             RESULT_FLUX, atol=1e-5*u.Unit(new_flux))
+    assert converted_spectrum.spectral_axis.unit == spectrum1d.spectral_axis.unit
+    assert converted_spectrum.uncertainty is None
+
+
+def test_no_spec_flux_unit_exp_not_none(specviz_app, spectrum1d):
+    label = "Test 1D Spectrum"
+    specviz_app.load_spectrum(spectrum1d, data_label=label)
+
+    new_flux = "erg / (s cm2 um)"
+
+    conv_func = uc.UnitConversion.process_unit_conversion
+    converted_spectrum = conv_func(specviz_app.app, spectrum=spectrum1d,
+                                   new_flux=new_flux)
+
+    assert_quantity_allclose(converted_spectrum.flux,
+                             RESULT_FLUX, atol=1e-5*u.Unit(new_flux))
+    assert_quantity_allclose(converted_spectrum.uncertainty.quantity.value,
+                             RESULT_UNCERTAINTY, atol=1e-5)
```

Comment:
```suggestion
                             RESULT_UNCERTAINTY, atol=1e-11)
```
The absolute tolerance value needs to be smaller, given that the values being compared are ~e-9

Reply:
Good catch!